src/github.com/getlantern/idletiming/idletiming_conn.go

```diff
@@ -100,12 +105,13 @@ func (c *IdleTimingConn) Read(b []byte) (int, error) {
 			n, err := c.conn.Read(b)
 			c.markActive(n)
 			totalN = totalN + n
-			timedOut := isTimeout(err)
-			if timedOut {
-				// Ignore timeouts when using deadline based on IdleTimeout
+			hitMaxDeadline := isTimeout(err) && !time.Now().Before(maxDeadline)
```

Comment:
Isn't the only timeout our own timeout (i.e. don't the connections not have a timeout)? What would another reason be? I guess we don't actually know what type of net.Conn it may be (might even be an idletiming_conn)?